### PR TITLE
Catelog-of-my-things

### PR DESCRIPTION
**Pull request by ikoote1**
### Project Update

**1.Team member # 1 - Imran Hossain**
- Create a Book class in a separate .rb file.
- Create a Label class with an association to the Item class (in a separate .rb file).
- All Book class properties visible in the diagram should be defined and set up in the constructor method.
- All Label class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
- add_item method in the Label class
- should take an instance of the Item class as an input
- should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object) 
- can_be_archived?() in the Book class
-  should override the method from the parent class
-  should return true if parent's method returns true OR if cover_state equals to "bad"
-  otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:
-  List all books
-  List all labels (e.g. 'Gift', 'New')
-  Add a book
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
-  books table (add all properties and associations from the parent Item class as table columns)
-  labels table

**2. Team member # 2  Santosh Konappanavar**

- Create MusicAlbum class in a separate .rb file.
- Create a Genre class with an association to the Item class (in a separate .rb file).
- All MusicAlbum class properties visible in the diagram should be defined and set up in the constructor method.
- All Genre class properties visible in the diagram should be defined and set up in the constructor method.
- Implement methods:
- add_item method in the Genre class
- should take an instance of the Item class as an input
- should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object)
- can_be_archived?() in the MusicAlbum class
- should override the method from the parent class
- should return true if the parent's method returns true AND if on_spotify equals true
- otherwise, it should return false
- Add unit tests for all implemented methods.
- The following options should be available:

      1. List all music albums
       2. List all genres (e.g 'Comedy', 'Thriller')
        3. Add a music album

- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created: music_albums table (add all properties and associations from the parent Item class as table columns) genres table

**3. Team Member # 3 Ikoote Rasuli**
- Create a Game class in a separate .rb file.
- Create an Author class with an association to the Item class (in a separate .rb file).
- All Game class properties visible in the diagram should be defined and set up in the constructor method.
- All Author class properties visible in the diagram should be defined and set up in the constructor method.
#### Implement methods:
- add_item method in the Author class
- should take an instance of the Item class as an input
- should add the input item to the collection of items
- should add self as a property of the item object (by using the correct setter from the item object)
- can_be_archived?() in the Game class
- should override the method from the parent class
- should return true if parent's method returns true AND if last_played_at is older than 2 years
- otherwise, it should return false
- Add unit tests for all implemented methods.
#### The following options should be available:
- List of games
- List all authors (e.g. 'Stephen King')
- Add a game
- All data should be preserved by saving collections in .json files.
- Create a schema.sql file with tables that will be analogical to the structure of the classes that you created:
- games table (add all properties and associations from the parent Item class as table columns)
- authors table

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  it 'checking for book publisher' do
+    expect(@book.publisher).to eq('publisher')
+  end
+
+  it 'checking for book coverstate' do
+    expect(@book.cover_state).to eq('coverstate')
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline spec/book_spec.rb:23]: - [ ] kindly test for the method `can_be_archived?` as it is missing in your test

File: item.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    rescue StandardError
+      publish_date = nil
+    end
+    if publish_date.nil?
+      false
+    else
+      (Date.today.year - publish_date.year) > 10
+    end
+  end
+end
```
Comment (VICTOR-CHUKWU) [inline item.rb:39]: - [ ] kindly add a test for this file, I can not see any test for it

File: README.md [hunk 3]
```diff
@@ -1,2 +1,171 @@
...
+# 📖 [Catalog of my things] <a name="about-project"></a>
+
+**[Catalog of my things]** is based on Ruby Language.
+In this project, we will create a console app that will help us to keep a record of different types of things our own: books, music albums, movies, and games. Everything will be based on the UML class diagram presented below. The data will be stored in JSON files but our will also prepare a database with tables structure analogical to our program's class structure.
+
+UML class daigram:
+<br/>
+<img src="catalog_of_my_things.png" width ="auto" height="auto" align="center">
+<br/>
+Video Presntation link:[click here](https://drive.google.com/file/d/19_cAoWVhG1qGclJdGupvGbvxVJmc9vG4/view?usp=sharing)
```
Comment (VICTOR-CHUKWU) [inline README.md:35]: - [ ] according to the project requirement All the members in the video should appear in the video and talk about the project., I can see only one person talking, all the members should be present and describe what they did in the project

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @imranhossainemi  and @ikoote1  and @Santosh-Konappanavar , 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: README.md [hunk 4]
```diff
@@ -1,2 +1,171 @@
...
+# 📖 [Catalog of my things] <a name="about-project"></a>
+
+**[Catalog of my things]** is based on Ruby Language.
+In this project, we will create a console app that will help us to keep a record of different types of things our own: books, music albums, movies, and games. Everything will be based on the UML class diagram presented below. The data will be stored in JSON files but our will also prepare a database with tables structure analogical to our program's class structure.
+
+UML class daigram:
+<br/>
+<img src="catalog_of_my_things.png" width ="auto" height="auto" align="center">
+<br/>
+Video Presntation link:[click here](https://drive.google.com/file/d/11iyYwzlnt4eh3poL4DpxWIh3PM0YZJLI/view?usp=sharing)
```
Comment (cynthiainga) [inline README.md:35]: - [ ] As mentioned by the previous reviewer all of you should appear in the video which means the camera is on 🙃, also when presenting the project make sure you show how it works. Everyone should explain their part, showing how it works in the console, and highlight a piece of the code that each one finds interesting or is proud of making 😁 😁

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @ikoote1, @imranhossainemi & @Santosh-Konappanavar, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.



 ## Highlights

- [ ] The app works as expected ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Data are preserved ✔️ 
- [ ] Each class has been created in a separate .rb file ✔️
- [ ] All tests are passing ✔️ 

 ## Required Changes ♻️

- [ ] Please kindly check the comment under the review 👇🏾 👇🏾 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @ikoote1, @imranhossainemi & @Santosh-Konappanavar, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ]  The app works as expected ✔️
- [ ] Descriptive PR ✔️
- [ ] Data are preserved ✔️
- [ ] Each class has been created in a separate .rb file ✔️
- [ ] All tests are passing ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can get the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**